extensions/warp-ipfs/src/store/message.rs

```diff
@@ -1,2369 +1,2837 @@
 
-        let peer_id = did_key.to_peer_id()?;
+                        if !list.contains(&self.keypair) {
+                            return Err(Error::ReactionDoesntExist);
+                        }
 
-        let event = ConversationEvents::NewConversation {
-            recipient: own_did.clone(),
-            settings,
-        };
+                        list.retain(|did| did != &(*self.keypair).clone());
+                        if list.is_empty() {
+                            e.remove();
+                        }
+                    }
+                    BTreeEntry::Vacant(_) => return Err(Error::ReactionDoesntExist),
+                };
 
-        let bytes = ecdh_encrypt(own_did, Some(did_key), serde_json::to_vec(&event)?)?;
-        let signature = sign_serde(own_did, &bytes)?;
+                message_document
+                    .update(&self.ipfs, &self.keypair, message, keystore.as_ref())
+                    .await?;
 
-        let payload = Payload::new(own_did, &bytes, &signature);
+                list.replace(message_document);
+                conversation.set_message_list(&self.ipfs, list).await?;
 
-        let peers = self.ipfs.pubsub_peers(Some(did_key.messaging())).await?;
+                self.set_document(conversation).await?;
 
-        if !peers.contains(&peer_id)
-            || (peers.contains(&peer_id)
-                && self
-                    .ipfs
-                    .pubsub_publish(did_key.messaging(), payload.to_bytes()?.into())
-                    .await
-                    .is_err())
-        {
-            warn!("Unable to publish to topic. Queuing event");
-            if let Err(e) = self
-                .queue_event(
-                    did_key.clone(),
-                    Queue::direct(
-                        convo_id,
-                        None,
-                        peer_id,
-                        did_key.messaging(),
-                        payload.data().to_vec(),
-                    ),
-                )
-                .await
-            {
-                error!("Error submitting event to queue: {e}");
+                _ = tx.send(MessageEventKind::MessageReactionRemoved {
+                    conversation_id,
+                    message_id,
+                    did_key: (*self.keypair).clone(),
+                    reaction: emoji.clone(),
+                });
             }
         }
 
-        self.event
-            .emit(RayGunEventKind::ConversationCreated {
-                conversation_id: convo_id,
-            })
-            .await;
+        let event = MessagingEvents::React {
+            conversation_id,
+            reactor: (*self.keypair).clone(),
+            message_id,
+            state,
+            emoji,
+        };
 
-        Ok(Conversation::from(&conversation))
+        self.publish(conversation_id, None, event, true).await
     }
 
-    pub async fn create_group_conversation(
+    pub async fn attach(
```

Comment:
Just a note: This function internals will likely change in a near future PR to cleanup parts that been there that is no longer needed.